Get libheif and pyheif to install correctly in Circle CI

I'm running into the issue from this thread: https://github.com/david-poirier-csn/pyheif/issues/1#issuecomment-578880808

https://app.circleci.com/pipelines/github/simonw/datasette-media/14/workflows/d5adc812-7681-44b0-a3bd-740bbc19bada/jobs/36

```
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/usr/local/include -I/home/circleci/project/venv/include -I/usr/local/include/python3.6m -c build/temp.linux-x86_64-3.6/_libheif_cffi.c -o build/temp.linux-x86_64-3.6/build/temp.linux-x86_64-3.6/_libheif_cffi.o
    build/temp.linux-x86_64-3.6/_libheif_cffi.c:1109:37: error: return type is an incomplete type
     static enum heif_color_profile_type _cffi_d_heif_image_handle_get_color_profile_type(struct heif_image_handle const * x0)
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    build/temp.linux-x86_64-3.6/_libheif_cffi.c: In function ‘_cffi_d_heif_image_handle_get_color_profile_type’:
    build/temp.linux-x86_64-3.6/_libheif_cffi.c:1111:10: warning: implicit declaration of function ‘heif_image_handle_get_color_profile_type’; did you mean ‘_cffi_d_heif_image_handle_get_color_profile_type’? [-Wimplicit-function-declaration]
       return heif_image_handle_get_color_profile_type(x0);
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 ...
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -38,8 +38,8 @@ jobs:
               && ./autogen.sh \
               && ./configure \
               && make -j4 \
-              && make install \
-              && ldconfig
+              && sudo make install \
+              && sudo ldconfig
             python3 -m venv venv
             . venv/bin/activate
             pip install -e .[test]
@@ -72,8 +72,8 @@ jobs:
               && ./autogen.sh \
               && ./configure \
               && make -j4 \
-              && make install \
-              && ldconfig
+              && sudo make install \
+              && sudo ldconfig
             python3 -m venv venv
             . venv/bin/activate
             pip install -e .[test]
@@ -102,8 +102,8 @@ jobs:
               && ./autogen.sh \
               && ./configure \
               && make -j4 \
-              && make install \
-              && ldconfig
+              && sudo make install \
+              && sudo ldconfig
             python3 -m venv venv
             . venv/bin/activate
             pip install -e .[test]

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -40,6 +40,7 @@ jobs:
               && make -j4 \
               && sudo make install \
               && sudo ldconfig
+            cd ..
             python3 -m venv venv
             . venv/bin/activate
             pip install -e .[test]
@@ -67,6 +68,7 @@ jobs:
       - run:
           name: install python dependencies
           command: |
+            pwd
             git clone https://github.com/strukturag/libheif.git libheif \
               && cd libheif \
               && ./autogen.sh \
@@ -74,6 +76,8 @@ jobs:
               && make -j4 \
               && sudo make install \
               && sudo ldconfig
+            cd ..
+            pwd
             python3 -m venv venv
             . venv/bin/activate
             pip install -e .[test]
@@ -104,6 +108,7 @@ jobs:
               && make -j4 \
               && sudo make install \
               && sudo ldconfig
+            cd ..
             python3 -m venv venv
             . venv/bin/activate
             pip install -e .[test]

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -33,7 +33,13 @@ jobs:
       - run:
           name: install python dependencies
           command: |
-            sudo apt-get install -y libheif-dev
+            git clone https://github.com/strukturag/libheif.git libheif \
+              && cd libheif \
+              && ./autogen.sh \
+              && ./configure \
+              && make -j4 \
+              && make install \
+              && ldconfig
             python3 -m venv venv
             . venv/bin/activate
             pip install -e .[test]
@@ -61,7 +67,13 @@ jobs:
       - run:
           name: install python dependencies
           command: |
-            sudo apt-get install -y libheif-dev
+            git clone https://github.com/strukturag/libheif.git libheif \
+              && cd libheif \
+              && ./autogen.sh \
+              && ./configure \
+              && make -j4 \
+              && make install \
+              && ldconfig
             python3 -m venv venv
             . venv/bin/activate
             pip install -e .[test]
@@ -85,7 +97,13 @@ jobs:
       - run:
           name: install python dependencies
           command: |
-            sudo apt-get install -y libheif-dev
+            git clone https://github.com/strukturag/libheif.git libheif \
+              && cd libheif \
+              && ./autogen.sh \
+              && ./configure \
+              && make -j4 \
+              && make install \
+              && ldconfig
             python3 -m venv venv
             . venv/bin/activate
             pip install -e .[test]

